tests/Functional/Schema/VariableDateTimePrecisionIntrospectionTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Schema;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\DB2Platform;
+use Doctrine\DBAL\Platforms\OraclePlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Schema\TableDiff;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Types\DateTimeType;
+use Doctrine\DBAL\Types\DateTimeTzType;
+use Doctrine\DBAL\Types\DateType;
+use Doctrine\DBAL\Types\TimeType;
+use Doctrine\DBAL\Types\Type;
+
+use function array_keys;
+use function count;
+use function get_class;
+use function sprintf;
+use function strpos;
+
+class VariableDateTimePrecisionIntrospectionTest extends FunctionalTestCase
+{
+    public function testTableIntrospection(): void
+    {
+        $tableName = 'test_time_prec_intros';
+
+        $platform = $this->connection->getDatabasePlatform();
+
+        if ($platform instanceof SqlitePlatform) {
+            self::markTestSkipped(sprintf("Platform %s doesn't support variable precision time", get_class($platform)));
+        }
+
+        $schemaManager = $this->connection->createSchemaManager();
+        $comparator    = $schemaManager->createComparator();
+
+        $table = new Table($tableName);
+
+        $doctrineTypes = array_keys(Type::getTypesMap());
+
+        foreach ($doctrineTypes as $type) {
```

Comment:
Flow control structures in tests, unless they are used for skipping tests, are a code smell. Each test scenario should be tested individually and unconditionally. Usually, table tests (or data providers) are a good alternative.

Reply:
Okay. I've updated with a data provider and set out the relevant types longhand.